doc/source/ray-core/fault-tolerance.rst

```diff
@@ -0,0 +1,67 @@
+Fault tolerance
+===============
+
+Ray is a distributed system, and a lot of failures can happen. Generally, it can
+be classified into two classes: 1) failure because of ray internally; 2) failure
+due to the applications. The first is likely to be triggered by node failure,
+network failure, or just bugs in Ray. The latter could happen because the user
+has bugs in their code or some external systems fail, and the failure is not
+handled well by the code. 
+Ray has built a lot of mechanisms to do its best to recover from the failure in
+both cases to ensure the distributed apps are fault tolerant.
+
+.. _fault-tolerance_ownership:
+
+Ownership model
+---------------
+
+To understand the failure model of Ray, we need to understand the ownership
```

Comment:
Hmm I'm not sure that we should include this section here. It seems a bit complicated to throw at users immediately and it's not immediately clear to a reader why they need to know about the ownership model.

Also, we mention the system vs application-level failures earlier, but it's a bit weird that we don't say anything else about how users should handle them. I feel like we need more of a high-level intro about what kinds of failures can be handled automatically or not.

Reply:
I feel it's hard to explain things if the user can't understand it. For example, the user create an object ref in an actor, and then pass it out with the method. Later the actor got killed, then the user can't access the object anymore.

If the user doesn't understand the ownership model, then for the cases where it can't be recovered, they will not understand.